src/param/param_unknown.rs

```diff
@@ -0,0 +1,64 @@
+use crate::param::param_header::ParamHeader;
+use crate::param::param_header::PARAM_HEADER_LENGTH;
+use crate::param::param_type::ParamType;
+use crate::param::Param;
+use bytes::{Bytes, BytesMut};
+use std::any::Any;
+use std::fmt::{Debug, Display, Formatter};
+
+/// This type is meant to represent ANY parameter for un/remarshaling purposes, where we do not have a more specific type for it.
+/// This means we do not really understand the semantics of the param but can represent it.
+///
+/// This is useful for usage in e.g.`ParamUnrecognized` where we want to report some unrecognized params back to the sender.
+#[derive(Clone, Debug, PartialEq)]
+pub struct ParamUnknown {
+    typ: u16,
+    value: Bytes,
+}
+
+impl Display for ParamUnknown {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "ParamUnknown( {} {:?} )", self.header(), self.value)
+    }
+}
+
+impl Param for ParamUnknown {
+    fn header(&self) -> ParamHeader {
+        ParamHeader {
+            typ: ParamType::Unknown {
+                param_type: self.typ,
+            },
+            value_length: self.value.len() as u16,
+        }
+    }
+
+    fn as_any(&self) -> &(dyn Any + Send + Sync) {
+        &*self
+    }
+
+    fn unmarshal(raw: &Bytes) -> crate::error::Result<Self>
+    where
+        Self: Sized,
+    {
+        let header = ParamHeader::unmarshal(raw)?;
+        let value = raw.slice(PARAM_HEADER_LENGTH..PARAM_HEADER_LENGTH + header.value_length());
```

Comment:
this will panic if `raw.len() < PARAM_HEADER_LENGTH + header.value_length()`. Are we sure raw will always be the right size here?

Reply:
Dunno, I copied this from `ParamHeartbeatInfo::unmarshal` impl. If this is a general concern the codebase should be vetted in it's entirety.

Reply:
I poked around a bit, this seems like it isn't really handled in any of the params unmarshal function. It could be checked in the `build_param` function though. Instead of just accessing the raw type, we could jusz unmarshal the full header and check that the bytes is long enough.

Reply:
thanks for looking into this! feel free to open an issue 👍

Reply:
I did open one. This seems to also affect chunk unmarshalling. I think this should be a whole new PR that should not block this one right?

Reply:
Yep